client/src/request.rs

```diff
@@ -267,4 +306,115 @@ impl SuccinctClient {
             Err(Error::msg("Failed to submit request to Succinct X API."))
         }
     }
+
+    /// Submit a request to the Succinct X API.
+    /// If in local prove mode, generates a local proof and returns the request_id after completion.
+    pub async fn submit_request(
+        &self,
+        chain_id: u32,
+        to: Address,
+        calldata: Bytes,
+        function_id: B256,
+        input: Bytes,
+    ) -> Result<String> {
+        if self.local_prove_mode {
+            return self.submit_local_request(
+                chain_id,
+                to,
+                calldata.clone(),
+                function_id,
+                input.clone(),
+            );
+        }
+
+        self.submit_platform_request(chain_id, to, calldata, function_id, input)
+            .await
+    }
+
+    /// If in local relay mode, ethereum_rpc_url and wallet must be provided. If you wish to submit
+    /// to your own gateway (ex. on a chain that doesn't have a canonical gateway), gateway_address
+    /// must be provided.
+    // TODO: Add support for hosted proving + local relaying.
+    pub async fn relay_proof(
+        &self,
+        request_id: String,
+        ethereum_rpc_url: Option<&str>,
+        wallet: Option<LocalWallet>,
+        gateway_address: Option<&str>,
+    ) -> Result<()> {
+        // If local mode, submit proof from local directory at proofs/output_{request_id}.json
+        if self.local_relay_mode {
+            let ethereum_rpc_url = ethereum_rpc_url
+                .expect("Ethereum RPC URL must be provided when relaying a proof in local mode.");
+            let wallet =
+                wallet.expect("Local wallet must be provided when relaying a proof in local mode.");
+
+            // Check if the proof file exists.
+            let proof_file = format!("{}/output_{}.json", LOCAL_PROOF_FOLDER, request_id);
+            if !Path::new(&proof_file).exists() {
+                return Err(Error::msg(format!(
+                    "Proof file {} does not exist.",
+                    proof_file
+                )));
+            }
+
+            // If it exists, attempt to submit the proof.
+            let proof_data = fs::read_to_string(proof_file)?;
+            let mut proof_json: serde_json::Value = serde_json::from_str(&proof_data)?;
+
+            // Strip alphanumeric characters from each of the proof_data fields (some are formatted
```

Comment:
Is this something that we could do when we save the proof so that we don't have to do this sort of hack?

Reply:
Yes